docs/3.x/enrich-js-tracker.md

```diff
@@ -0,0 +1,189 @@
+---
+category: Develop
+title: JS Tracker
+---
+# Enriching the Matomo JavaScript Tracker
+
+You can add additional code to the `matomo.js` JavaScript tracker. This allows you for example to:
+
+* configure the Matomo JavaScript tracker in a certain way to ensure the same behaviour is applied across all sites. For example enforcing that cookies are disabled on all sites.
+* track additional information about the visitor such as tracking additional browser plugins or other browser or device data.  
+* add additional analytics features to the tracker such as JS error tracking
+
+## Enriching the tracker:
+
+To enrich the tracker create a file named `tracker.js` in your Matomo plugin directory. The base template for this file looks like this:
+
+```js
+(function () {
+
+    function init() {
+        // Add your plugin code here. For example to enforce disabling cookies:
+        window._paq = window._paq || [];
+        winodw._paq.push(['disableCookies']);
+    }
+
+    if ('object' === typeof window.Piwik) {
+        init();
+    } else {
+        // tracker might not be loaded yet
+        if ('object' !== typeof window.piwikPluginAsyncInit) {
+            window.piwikPluginAsyncInit = [];
+        }
+
+        window.piwikPluginAsyncInit.push(init);
+    }
+
+})(); 
+```
+
+### Updating the JS tracker
+
+In order to test your tracker integration, execute the following console command within your root Matomo directory.
+
+```bash
+./console custom-piwik-js:update --ignore-minified
```

Comment:
Should we rename that event for Matomo 4 to `custom-tracker-js:update` ? 🤔

Reply:
We'll need to adjust this in the 4.x-dev specific docs later.